backend/routers/spiritual.py

```
@@ -789,4 +907,136 @@ def get_default_lord(house_index: int) -> str:
+            WHERE email = $2
+        """, json.dumps(complete_profile), user_email)
+        await conn.close()
+        
+        return {
+            "success": True,
+            "complete_profile": complete_profile
+        }
+        
+    except json.JSONDecodeError as e:
+        logger.error(f"JSON decode error generating birth chart: {e}")
+        raise HTTPException(status_code=500, detail="Invalid birth chart data format")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)